### PR TITLE
[Fixes #3433] Invalid media type 'text/plain; charset=utf-8'

### LINKED ISSUE #3433: Invalid media type 'text/plain; charset=utf-8'
In a Controller action:

    return File(stream, "text/plain; charset=utf-8");

throws `'System.FormatException'` in `Microsoft.Net.Http.Headers.dll` with message `Invalid media type 'text/plain; charset=utf-8'`

### PR DESCRIPTION
**Pull request by kichalla**
@rynowak

### CODE REVIEW

File: samples/MvcSample.Web/project.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "commands": {
-    "web": "Microsoft.AspNet.Hosting server=Microsoft.AspNet.Server.WebListener server.urls=http://localhost:5001",
+    "web": "Microsoft.AspNet.Server.Kestrel",
```
Comment (dougbu) [inline samples/MvcSample.Web/project.json]: This project is in the midst of a fiery death (see #3574). Don't bother with any changes here.

File: src/Microsoft.AspNet.Mvc.Core/FileResult.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public abstract class FileResult : ActionResult
         /// </summary>
         /// <param name="contentType">The Content-Type header of the response.</param>
         protected FileResult(string contentType)
-            : this(new MediaTypeHeaderValue(contentType))
+            : this(MediaTypeHeaderValue.Parse(contentType))
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/FileResult.cs]: This constructor overload doesn't seem to be used except in tests. Can we remove it? If not, should test using the `EmptyFileResult`.

**Comment by dougbu** [comment]
:watch:

File: samples/MvcSample.Web/project.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "commands": {
-    "web": "Microsoft.AspNet.Hosting server=Microsoft.AspNet.Server.WebListener server.urls=http://localhost:5001",
```
Comment (rynowak) [inline samples/MvcSample.Web/project.json]: why this?

File: samples/MvcSample.Web/wwwroot/web.config [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <system.webServer>
+    <handlers>
+      <add name="httpPlatformHandler" path="*" verb="*" modules="httpPlatformHandler" resourceType="Unspecified" />
+    </handlers>
+    <httpPlatform processPath="%DNX_PATH%" arguments="%DNX_ARGS%" forwardWindowsAuthToken="false" startupTimeLimit="3600" />
+  </system.webServer>
```
Comment (rynowak) [inline samples/MvcSample.Web/wwwroot/web.config]: ???

**Comment by rynowak** [comment]
Do a CTRL+F - are there any other uses of `new MediaTypeHeaderValue()`

File: src/Microsoft.AspNet.Mvc.Core/FileResult.cs (hunk 2 above)
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/FileResult.cs]: :+1:

File: samples/MvcSample.Web/project.json (hunk 3 above)
Comment (kichalla) [inline samples/MvcSample.Web/project.json]: This was a change automatically made by VS(i left it thinking that it is anyways correct...as one should be able to do Ctrl + F5 on a sample in VS)...but this is now not required due to the sand box changes...

File: samples/MvcSample.Web/wwwroot/web.config (hunk 4 above)
Comment (kichalla) [inline samples/MvcSample.Web/wwwroot/web.config]: yeah...same comment as earlier

Comment (rynowak) [inline samples/MvcSample.Web/wwwroot/web.config]: This project was removed. Remember to rebase.

**Comment by rynowak** [comment]
:shipit: after reverting changes to `samples/MvcSample.Web`. That project was removed.

**Comment by kichalla** [comment]
Updated the PR based on feedback and also Content helper changes in Controller.cs
@rynowak @dougbu

File: test/Microsoft.AspNet.Mvc.Core.Test/ContentResultTest.cs [hunk 5]
```diff
@@ -195,5 +255,28 @@ private static HttpContext GetHttpContext()
...
+
+            public IActionResult Content_WithInvalidCharset()
+            {
+                return Content("Hello!!", "text/xml; charset=invalid; p1=p1-value");
+            }
+
+            public IActionResult Content_WithEncodingInCharset_AndEncodingParameter()
+            {
+                return Content("Hello!!", "text/xml; charset=invalid; p1=p1-value", Encoding.ASCII);
+            }
```
Comment (rynowak) [inline test/Microsoft.AspNet.Mvc.Core.Test/ContentResultTest.cs]: These are tests for the controller methods, not for `ContentResult`. These should go somewhere else.

File: samples/MvcSandbox/project.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ﻿{
   "commands": {
-    "web": "Microsoft.AspNet.Hosting server=Microsoft.AspNet.Server.WebListener server.urls=http://localhost:5001",
+    "web": "Microsoft.AspNet.Server.Kestrel",
```
Comment (dougbu) [inline samples/MvcSandbox/project.json]: Does this work without a `hosting.json` file? If not, please `git add` that file to this commit.

**Comment by davidfowl** [comment]
Sigh, strongly typed headers...

**Comment by dougbu** [comment]
:shipit: once you get @rynowak's squirrel.

File: samples/MvcSandbox/project.json (hunk 6 above)
Comment (kichalla) [inline samples/MvcSandbox/project.json]: yes, it works without a `hosting.json` file.

**Comment by kichalla** [comment]
:up: :date: @rynowak

**Comment by rynowak** [comment]
@davidfowl :trollface: 

:shipit:

**Comment by kichalla** [comment]
31e42ee31286fe115c75336343a6cef68729f4d7

**Pull request closed without merging**